adminapi/src/main/java/io/minio/admin/MinioAdminClient.java

```diff
@@ -613,6 +619,257 @@ public Message getServerInfo() throws IOException, NoSuchAlgorithmException, Inv
     }
   }
 
+  /**
+   * Creates a new service account belonging to the user sending.
+   *
+   * <pre>Example:{@code
+   * // Assume policyJson contains below JSON string;
+   * // {
+   * //     "Statement": [
+   * //         {
+   * //             "Action": "s3:GetObject",
+   * //             "Effect": "Allow",
+   * //             "Principal": "*",
+   * //             "Resource": "arn:aws:s3:::my-bucketname/myobject*"
+   * //         }
+   * //     ],
+   * //     "Version": "2012-10-17"
+   * // }
+   * //
+   * }</pre>
+   *
+   * @param accessKey Access key.
+   * @param targetUser Target user.
+   * @param secretKey Secret key.
+   * @param policy Policy as JSON string .
+   * @param name Service account name.
+   * @param description Description for this access key.
+   * @param comment Deprecated: use description instead.
+   * @param expiration Expiry time , Example : 2023-12-02T15:04:05Z.
+   * @return Service account info for the specified accessKey.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws IOException thrown to indicate I/O error on MinIO REST operation.
+   * @throws InvalidCipherTextException thrown to indicate data cannot be encrypted/decrypted.
+   */
+  public Credentials addServiceAccount(
+      @Nonnull String targetUser,
+      @Nullable String name,
+      @Nonnull String secretKey,
+      @Nonnull String accessKey,
+      @Nullable String policy,
+      @Nullable String description,
+      @Nullable ZonedDateTime expiration,
+      @Nullable String comment)
+      throws NoSuchAlgorithmException, InvalidKeyException, IOException,
+          InvalidCipherTextException {
+    if (targetUser == null || targetUser.isEmpty()) {
+      throw new IllegalArgumentException("target user must be provided");
+    }
+    if (targetUser.length() > 32) {
```

Comment:
`targetUser` doesn't have any validation like this; but `name` has to be validated like
```go
	if len(name) > 32 {
		return errors.New("name must not be longer than 32 characters")
	}
	if !serviceAcctValidNameRegex.MatchString(name) {
		return errors.New("name must contain only ASCII letters, digits, underscores and hyphens and must start with a letter")
	}
```